.dialyzer_ignore.exs

```diff
@@ -0,0 +1,18 @@
+[
+    ~r/The test \d* =:= nil can never evaluate to 'true'/
```

Comment:
This is the offending line (when `depth` is used in the `feature` macro):
https://github.com/exercism/elixir-analyzer/blob/cc85fea1abaa1a113da7d787faefff1fc27e8217/lib/elixir_analyzer/exercise_test/feature/compiler.ex#L107

It fails because after unquoting in produces code `1 in [nil, depth]`. No idea how to fix this.

Reply:
The goal I wanted to achieve in this PR is to remove the requirement to add this module attribute to every module that uses `ElixirAnalyzer.ExerciseTest` so that the process of writing an analyzer extension is a bit simpler.

If it was my project only, I would only remove this module attribute from everywhere and not do anything else. But since you're saying ElixirLS is a problem for you, and it will run dialyzer for every project, I think it would be nice to have a CI check that the dialyzer passes, even if the only advantage of that is that you (and other VSC users?) won't get bothered by warnings :).

So in my opinion, the PR in its current state satisfies that goal and I definitely don't need any rewrites of all the macros just to satisfy dialyzer 😅

Reply:
Yea, without that attribute in each actual file, it fills my ide with "problems":

![Screenshot from 2021-04-24 19-46-09](https://user-images.githubusercontent.com/8953691/115977465-c44a8580-a535-11eb-9649-b33afae0ebce.png)

So if we could leave that in the test files I would appreciate it.  All other changes are fine.

Reply:
Thank you for the screenshot, it made me realize where to look in VSC to see if my fix works. I can totally understand why you don't want to have those "problems" reported and I think I _FINALLY_ have a solution that works :).

However, I couldn't keep `@dialyzer generated: true` and also keep my other changes because this module attribute crashes the task `mix dialyzer`, I am 99% of that. With this attribute, the task `mix dialyzer` would always fail on CI.

*TL;DR:*
I realized that we cannot use dialyxir's ignore file to ignore the warnings because ElixirLS doesn't use dialyxir, it uses dialyzer directly.

I was able to ignore the specific warnings with correct `@dialyzer` annotations so that now ElixirLS doesn't report those "problems" in VSC, and also `mix dialyzer` passes.

I think we both have now what we wanted 🤝